src/bin/scenarist-desktop/ManagementLayer/Scenario/ScenarioManager.cpp

```diff
@@ -1214,3 +1228,26 @@ BusinessLogic::ScenarioDocument* ScenarioManager::workingScenario() const
 {
     return m_workModeIsDraft ? m_scenarioDraft : m_scenario;
 }
+
+void ScenarioManager::lockSceneNumbers()
+{
+    bool allowedLock = true;
+
+    if ((m_workModeIsDraft && m_fixedScenesDraft)
+            || (!m_workModeIsDraft && m_fixedScenes)) {
+        QDialogButtonBox::StandardButton result = QLightBoxMessage::question(m_view, tr("Unlock?"),
+        tr("Do you want to unlock or lock?"),
```

Comment:
title - Changing scenes numbers locking
text - Do you want to unlock scenes numbers or lock again?